builtin/providers/azurerm/resource_arm_loadbalancer_backend_address_pool.go

```diff
@@ -0,0 +1,207 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/arm/network"
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/jen20/riviera/azure"
+)
+
+func resourceArmLoadbalancerBackendAddressPool() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmLoadbalancerBackendAddressPoolCreate,
+		Read:   resourceArmLoadbalancerBackendAddressPoolRead,
+		Delete: resourceArmLoadbalancerBackendAddressPoolDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": {
+				Type:      schema.TypeString,
+				Required:  true,
+				ForceNew:  true,
+				StateFunc: azureRMNormalizeLocation,
+			},
+
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"loadbalancer_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"backend_ip_configurations": {
```

Comment:
Is the convention for all of these not to set as a _singular_ name such that you can use the internal blocks? Such as `aws_security_group`